google_api_lib/google_api_client.py

```diff
@@ -0,0 +1,214 @@
+                        'startRowIndex': 0,
+                        'endRowIndex': 4 + len(feeders),
+                        'startColumnIndex': 0,
+                        'endColumnIndex': 5,
+                    },
+                    'rows': [
+                        {'values': [{'userEnteredValue': {'stringValue': 'THIS PAGE IS AUTOGENERATED AND WILL BE OVERWRITTEN WHEN THERE ARE PUZZLE UPDATES.'}}]},
+                        {'values': [{'userEnteredValue': {'stringValue': 'DO NOT MAKE CHANGES HERE; COPY INFO TO ANOTHER SHEET FIRST.'}}]},
+                        {},
+                        {'values': [
+                            {'userEnteredValue': {'stringValue': 'Puzzle Link'}, 'userEnteredFormat': {'textFormat': {'bold': True}}},
+                            {'userEnteredValue': {'stringValue': 'Puzzle Name'}, 'userEnteredFormat': {'textFormat': {'bold': True}}},
+                            {'userEnteredValue': {'stringValue': 'Answer'}, 'userEnteredFormat': {'textFormat': {'bold': True}}},
+                            {'userEnteredValue': {'stringValue': 'Sheet'}, 'userEnteredFormat': {'textFormat': {'bold': True}}},
+                            {'userEnteredValue': {'stringValue': 'Slack'}, 'userEnteredFormat': {'textFormat': {'bold': True}}},
+                        ]},
+                    ] + [
+                        {'values': [
+                            {'userEnteredValue': {'formulaValue': '=HYPERLINK("%s", "puzzle")' % puzzle.url}},
+                            {'userEnteredValue': {'stringValue': puzzle.name}},
+                            {'userEnteredValue': {'stringValue': puzzle.answer}, 'userEnteredFormat': {'textFormat': {'fontFamily': 'Roboto Mono'}}},
+                            {'userEnteredValue': {'formulaValue': '=HYPERLINK("%s", "sheet")' % puzzle.sheet}},
+                            {'userEnteredValue': {'formulaValue': '=HYPERLINK("%s/app_redirect?channel=%s", "channel")' % (settings.SLACK_BASE_URL, puzzle.channel)}},
+                        ]} for puzzle in feeders
+                    ],
+                },
+            }, {
+                'autoResizeDimensions': {
+                'dimensions': {
+                    'sheetId': sheet_id,
+                    'dimension': 'COLUMNS',
+                    'startIndex': 1,
+                    'endIndex': 3,
+                },
+            },
+            }]
+        }
+
+        self._sheets_service.spreadsheets().batchUpdate(
+            spreadsheetId=spreadsheet_id,
+            body=body
+        ).execute(http=http)
+
+    @staticmethod
+    def update_meta_sheet_feeders(meta_puzzle):
+        '''
+        Updates the input meta puzzle's spreadsheet with the
+        latest feeder puzzle info
+        '''
+        if not meta_puzzle.is_meta:
+            return
+
+        client = GoogleApiClient.getInstance()
+        if not client:
+            return
+
+        # update meta sheet in separate thread
```

Comment:
This is probably fine, but usually this would be handled with a task queue: https://djangopackages.org/grids/g/workers-queues-tasks/

Instructions for RQ / Celery on Heroku: https://devcenter.heroku.com/categories/background-jobs-in-python

Reply:
Yeah, I saw a lot of mentions about that but didn't feel like introducing another library, learning it, and setting it up. Using threads seemed easier and more lightweight :)

Reply:
Using RQ doesn't seem too difficult. I'll add a TODO to switch to that (or something else) later. Ref: https://devcenter.heroku.com/articles/python-rq

Though I think a DB-based queue is preferable since we're already running a DB (and RQ requires running a Redis instance). So maybe something like Django Background Tasks is preferable: https://django-background-tasks.readthedocs.io/en/latest/